sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -93,35 +98,77 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
    * MSFAA received on the database with the
    * information received. If the information was already received
    * the record will not be updated.
-   * @param msfaaNumber MSFAA number
-   * @param dateSigned date in which the borrower indicated the MSFAA was signed
-   * @param serviceProviderReceivedDate date in which the MSFAA was received by/resolve from CanadaPost/Kiosk
-   * @returns update result. Only one row is supposed to be affected.
+   * @param msfaaNumber MSFAA number.
+   * @param offeringIntensity associated offering intensity.
+   * @param dateSigned date in which the borrower indicated the MSFAA was signed.
+   * @param serviceProviderReceivedDate date in which the MSFAA was received by/resolve from CanadaPost/Kiosk.
    */
-  async updateReceivedFile(
+  async updateReceivedRecord(
     msfaaNumber: string,
+    offeringIntensity: OfferingIntensity,
     dateSigned: Date,
     serviceProviderReceivedDate: Date,
-  ): Promise<UpdateResult> {
+  ): Promise<void> {
     if (!dateSigned || !serviceProviderReceivedDate) {
       throw new Error(
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
-    return this.repo.update(
-      {
-        msfaaNumber: msfaaNumber,
-        dateSigned: IsNull(),
-        serviceProviderReceivedDate: IsNull(),
+    // Check if the msfaa number needs to be reactivated or simply updated with a signed date. The msfaa record will need to reactivated if it is present in the database as cancelled.
+    const retrievedMSFAARecord = await this.repo.findOne({
+      select: {
+        id: true,
+        student: { id: true },
+        referenceApplication: { id: true },
+        cancelledDate: true,
+        dateSigned: true,
+        offeringIntensity: true,
       },
-      {
-        dateSigned: getISODateOnlyString(dateSigned),
-        serviceProviderReceivedDate: getISODateOnlyString(
-          serviceProviderReceivedDate,
-        ),
+      relations: {
+        student: true,
+        referenceApplication: true,
       },
-    );
+      where: {
+        msfaaNumber,
+        offeringIntensity,
+      },
+    });
+    const systemUser = await this.systemUsersService.systemUser();
+    // If there is a retrievedMSFAARecord, check if the msfaa number was cancelled earlier. If yes, reactivate it.
+    if (retrievedMSFAARecord?.cancelledDate) {
```

Comment:
Here the `cancelledDate` is potentially null not the `retrievedMSFAARecord` hence the null check in `retrievedMSFAARecord?.cancelledDate` would not be applicable.
Before line 136 we can add a specific check to ensure that the `findOne` returned a record, if not, we can throw an exception stating that the MSFAA number was not found.

Reply:
I do agree with @andrewsignori-aot and this has been already added as a comment https://github.com/bcgov/SIMS/pull/2010#discussion_r1230090168.